### DISCUSSION TITLE
What is the purpose of turbo's postinstall script and should I allow it to run?

### DISCUSSION THREAD

**Discussion by jraoult** [vercel/turborepo#3932] [Help] [answered]
I'm using `yarn` with the [`enableScript` option](https://yarnpkg.com/configuration/yarnrc#enableScripts) set to `false` so that, by default, we don't run any script third-party script.

I haven't noticed any issue so far with that setup and turbo, but I wonder what is turbo's `postinstall` script doing, and should I allow it to run?

**Comment by tknickman** [comment, marked as answer]
Here is the script: https://github.com/vercel/turbo/blob/main/packages/turbo/install.js

We use the postinstall to provide a bit of extra insurance due to the installation method of turbo. Because we install platform specific binaries via optional dependencies (https://github.com/vercel/turbo/blob/main/packages/turbo/package.json#L21-L28), the postinstall script helps us validate that this process worked as expected, and handles some common error cases. 

As you noted, it should work fine without this script as long as you don't fall into any of the corner cases (you're allowing optional dependencies and running on a supported OS)
